Added support for exporting/importing XLIFF files (1.2/2.0)

yarick2000 opened the pull request:
- Adds 2 new export format XLIFF 1.2 and XLIFF 2.0 for collections with translations fields.
- Collections with multiple translations fields are supported.  In this case user will need to select one of them for export.
  Also user will need to specify a language for exporting items. Language selected from the list of languages supported by the 
  specified translation field. In case if there is only one translation field in current collection,  it will be automatically used for 
  exporting .
- Adds new importing functionality (right now only XLIFF 1.2/2.0 supported) for collections with translations fields. 
- Collections with multiple translations fields are also supported. User may skip selection of the language for importing and use the one that specified in imported file.

Also, the following things were done:
- new POST endpoint for /collection has been created for implementing importing functionality
- new XLIFF service has been created in API. Its responsible for uploading/downloading XLIFF content to/from specific collection 
- Query type and query related utils were updated with 'xliff' and 'xliff2' values for export prop
- Query type has been extended with a new prop called 'optional' for additional parameters that can be used by endpoints for bringing information in addition to what is already supported by Query type.
- new component were created:
a) import-sidebar-detail - implementing UI part for importing functionality.  
b) translation-field-select - allows to select a specific translation field from the list of translations fields related to specific collection
c) language-select - allows to select a specific language from the list of languages associated with specific translation field of specific collection
d) file-select - allows to select a single file from a local device

**Example of the export file for XLIFF 1.2 (English):**
`<xliff xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 http://docs.oasis-open.org/xliff/v1.2/os/xliff-core-1.2-strict.xsd" xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
  <file original="articles_translations" datatype="plaintext" source-language="en-US">
    <body>
      <trans-unit id="b2d76785-2f4d-421b-830c-d80bd7ba029e.title">
        <source>Article Title</source>
      </trans-unit>
      <trans-unit id="b2d76785-2f4d-421b-830c-d80bd7ba029e.content">
        <source>&lt;p&gt;This is some text to &lt;strong&gt;translate&lt;/strong&gt;&lt;/p&gt;</source>
      </trans-unit>
    </body>
  </file>
</xliff>`

**Example of the import file for XLIFF 1.2 (Russian):**
`<?xml version="1.0" encoding="UTF-8"?>
<xliff xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 http://docs.oasis-open.org/xliff/v1.2/os/xliff-core-1.2-strict.xsd" xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
  <file original="articles_translations" datatype="plaintext" source-language="en-US" target-language="ru-RU">
    <body>
      <trans-unit id="b2d76785-2f4d-421b-830c-d80bd7ba029e.title">
        <source>Article Title</source>
        <target>Заголовок статьи</target>
      </trans-unit>
      <trans-unit id="b2d76785-2f4d-421b-830c-d80bd7ba029e.content">
        <source>&lt;p&gt;This is some text to &lt;strong&gt;translate&lt;/strong&gt;&lt;/p&gt;</source>
        <target>&lt;p&gt;Это текст для &lt;strong&gt;перевода&lt;/strong&gt;&lt;/p&gt;</target>
      </trans-unit>
    </body>
  </file>
</xliff>`

Unit id contains of two parts separated by '.' First one is primary key value for the collection item and second is a name of the translatable field. The namespace contains the name of the translations collection.

yaroslav-vysotsky committed:
ARC-4532 (not finished):

* Added new XLIFF format to export
* Added 'translations to list of supported field types
* Added XLIFF label to en-US
* Added language parameter to allowed query string params
* Started to work on XLIFF export on API side

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

yaroslav-vysotsky committed:
ARC-4532:

* Added code for obtaining translations data from translations collection
* Added 'translations' to the list of interface constants

yaroslav-vysotsky committed:
ARC-4532:

* Added xliff 3rd party module for xliff conversions
* Finished export functionality

yaroslav-vysotsky committed:
ARC-4532:

* removed unnecessary casting and added some formatting

yaroslav-vysotsky committed:
ARC-4532:

* Added error handling for case when translations relationship cannot be found

yaroslav-vysotsky committed:
ARC-4532:

* Converted XLIFF functionality to a service
* Added support for XLIFF 2.0

yaroslav-vysotsky committed:
ARC-4532:

* Fixed issue with initialization of export formats dropdown

yaroslav-vysotsky committed:
WIP on ARC-4532: 830d1ca7c ARC-4532:

yaroslav-vysotsky committed:
index on ARC-4532: 830d1ca7c ARC-4532:

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

# Conflicts:
#	api/src/utils/apply-query.ts
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
Added support for exporting items in XLIFF 1.2/2.0 format

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

# Conflicts:
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
Merge remote-tracking branch 'origin/main' into fix/4658-export

# Conflicts:
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
fix/4658:

* Removed unused import from sanitize-query

yaroslav-vysotsky committed:
ARC-4532:

Removed unused import from sanitize-query

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into fix/4658-export

yaroslav-vysotsky committed:
WIP on ARC-4532: 6be4c9ffc ARC-4532:

yaroslav-vysotsky committed:
index on ARC-4532: 6be4c9ffc ARC-4532:

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

yaroslav-vysotsky committed:
Merge commit '92b645929044d387126b6a188261de523c017134' into task/ARC-4532

yaroslav-vysotsky committed:
ARC-4532:

Formatted code

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into fix/4658-export

yarick2000 committed:
Fixed merge artifacts

yarick2000 committed:
Fixed merge artifact

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

yaroslav-vysotsky committed:
ARC-4567:

* Created new section component - import-sidebar-detail

yaroslav-vysotsky committed:
Fixed file name extension

yaroslav-vysotsky committed:
Merge branch 'fix/4658-export' into task/ARC-4532

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into task/ARC-4567

yaroslav-vysotsky committed:
ARC-4567:

* Refactored file-upload component that was orginally copied from the file interface

yaroslav-vysotsky committed:
Merge branch 'main' into fix/4658-export

# Conflicts:
#	package-lock.json

yaroslav-vysotsky committed:
ARC-4567:

* Renamed file-upload to file-select
* Added onLoad event to file-select component

yaroslav-vysotsky committed:
ARC-4567:

* Added language-select component

yaroslav-vysotsky committed:
ARC-4567:

* Added translation-field-select component

yaroslav-vysotsky committed:
ARC-4532:

* Replaced custom methods with lodash ones
* Improved validation for null/undefined values

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into fix/4658-export

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	package-lock.json

yaroslav-vysotsky committed:
ARC-4567

* Fixed incorrect import of the method

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

yaroslav-vysotsky committed:
ARC-4567:

* Fixed some UI validation logic related to visibility of some components in in import-sidebar-detail
* Made items computed in translation-field-select

yaroslav-vysotsky committed:
ARC-4567:

* Introduced new Collections enpoint - /collections/:collection/import/:format
* Strarted to work on importData method

yaroslav-vysotsky committed:
ASRC-4567:

* Added new POST endpoint to collections controller
* Added fromXliff method (not finished yet) to xliff service
* Fixed small selection issue in file-select component
* Updated import-sidebar-detail component with new route for importing file

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into task/ARC-4567

# Conflicts:
#	api/src/services/xliff.ts

yaroslav-vysotsky committed:
ARC-4567:

* Moved import logic to a separate function (import)
* Finished import functionality

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
ARC-4567:

* Added call to log in case of error
* Refactored import so it makes less calls to DB
* Added refresh event to import-sidebar-detail component
* Added 'importing' state to import-sidebar-detail component

yaroslav-vysotsky committed:
ARC_4567:

* Moved file uploading into separate handler

yaroslav-vysotsky committed:
ARC-4567:

* Moved parmeters to the form from query string
* Removed unneeded progressUpload event
* Added notification in case of successfull import

yaroslav-vysotsky committed:
ARC-4567:

* Removed 'language' prop from Query type and replaced it with more generic 'optional' parameter, that allows to use multiply additional params if needed without touching the main set of props.
* Updated export-sidebar-detail component accordingly

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4532

# Conflicts:
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
ARC-4567:

* Removed 'language' prop from Query type and replaced it with more generic 'optional' parameter, that allows to use multiply additional params if needed without touching the main set of props.
* Updated export-sidebar-detail component accordingly

# Conflicts:
#	api/src/middleware/respond.ts
#	api/src/services/xliff.ts

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4532' into task/ARC-4567

# Conflicts:
#	api/src/middleware/respond.ts
#	api/src/services/xliff.ts

yaroslav-vysotsky committed:
ARC-4567:

* Added language and translations field selection to export-sidebar-detail component
* Updated API code accordingly

yaroslav-vysotsky committed:
ARC-4567:

* Refactored export-export-sidebar-detail and language-select component - removed 'value' prop and introduced model binding

yaroslav-vysotsky committed:
ARC-4567:

* Added error handling for import

yaroslav-vysotsky committed:
ARC-4567

* Fixed some visibility  logic in import-sidebar-detail component
* Returned back code that was accidentally removed in collections controller

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

# Conflicts:
#	app/src/modules/collections/routes/collection.vue

yaroslav-vysotsky committed:
ARC-4567:

* Made 'accept' attribute for file-select dynamic

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	app/src/modules/collections/routes/collection.vue

yaroslav-vysotsky committed:
Merge branch 'main' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	api/src/controllers/collections.ts
#	api/src/utils/sanitize-query.ts

yaroslav-vysotsky committed:
ARC-4567:

* renamed some variables in import method to a more relevant ones
* modified key processing in order to cover custom string keys that may contain '.'

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'main' into fix-4658

# Conflicts:
#	docs/guides/accessing-data.md

yaroslav-vysotsky committed:
ARC-4567:

* modified items search in import, used string key comparison
* replaced deprecated call to create method with createMany

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

rijkvanzanten committed:
Merge branch 'main' into fix-4658

rijkvanzanten commented on api/src/controllers/collections.ts:
```diff
@@ -1,13 +1,49 @@
+	busboy.on('file', async (fieldname, fileStream, filename, encoding, mimetype) => {
+		fileCount++;
+		const { format } = res.locals.fields;
+		if (['xliff', 'xliff2'].includes(format)) {
+			if (fileCount > 1) {
+				busboy.emit('error', new InvalidPayloadException(`Only one import file can be used for XLIFF format.`));
+			}
+			let content = '';
+			fileStream.on('data', (d) => (content += d)).on('end', () => (res.locals.data = content));
+		} else {
```
I think we'll spin this into a bigger `/import` endpoint/system, as per https://github.com/directus/directus/issues/2927. I'd rather not handle this under the `/collections` endpoint, as that will cause confusion between the functionality (creating a new collection vs reading the file and injecting the content into the DB)

yarick2000 replied:
Right now the export/import of XLIFF content happens from parent collection in terms of UI. The functionality available if there is one or more fields with type 'translations' in collection. I was thinking that this is will be the more natural way neither then adding this functionality right to translations collection, but maybe I'm not right. Translators (from my experience) usually operating with the original entities, not the translated items. Also, in case if they will work with exporting from translations collection they will need to make some filtering for specific language before making export.  Now, what if they selected two or more items that related to different languages, so how to export them? We can use exactly the same interface with languages dropdown in order to filter them out, but it will be kind of duplication of UX behavior - they will need to select language twice then, once for filtering items and second time before exporting...

rijkvanzanten replied:
> I was thinking that this is will be the more natural way neither then adding this functionality right to translations collection, but maybe I'm not right. 

Yes you're absolutely right from a UX perspective in the app. That being said, I think it's technically easier to reason about if the export itself actually happens on the related translations collections, instead of through the o2m field

yarick2000 replied:
Are, so you mean to refactor XliffService so it will work with translations collection directly instead of using the parent one as a parameter, right?

rijkvanzanten replied:
Yes 👍🏻

yarick2000 replied:
Ok, will try to do this as well.

rijkvanzanten commented on api/src/controllers/collections.ts:
```diff
@@ -101,6 +137,37 @@ router.patch(
 	respond
 );
 
+router.post(
+	'/:collection',
+	multipartHandler,
```
This shouldn't be under `/collections/:collection`, as per the previous comment

yarick2000 replied:
So which route you will suggest? Could it be /collections/:collection/import?

rijkvanzanten replied:
`/collections` is used for schema management, so that's out of the question to me. I'm thinking either `POST /items/:collection` with a file, or `POST /items/:collection/import`

yarick2000 replied:
Ok, will fix it

rijkvanzanten commented on api/src/middleware/respond.ts:
```diff
@@ -67,6 +68,22 @@ export const respond: RequestHandler = asyncHandler(async (req, res) => {
 				return stream.pipe(json2csv).pipe(res);
 			}
 		}
+		if (['xliff', 'xliff2'].includes(req.sanitizedQuery.export)) {
```
I think we'll end up moving this whole logic (including json/csv) into a dedicated ExportService (as part of https://github.com/directus/directus/issues/2927)

yarick2000 replied:
Do you want me to work on that in scope of this task? Or you want me to finish first with that one and then start to work on https://github.com/directus/directus/issues/2927? I, honestly, not sure if I will have enough time to work on big tasks within the upcoming month, but maybe later...

rijkvanzanten replied:
No that's ok! I'll pick up the work on #2927 and get that across the finish line. Then we can update this PR to fit within the finalized structure for that 🙂

yarick2000 replied:
Ok

rijkvanzanten commented on api/src/services/xliff.ts:
```diff
@@ -0,0 +1,277 @@
+	format: string;
+};
+
+export class XliffService {
```
Sweet! This looks really full featured 🙂

rijkvanzanten commented on api/src/services/xliff.ts:
```diff
@@ -0,0 +1,277 @@
+import { FieldsService } from './fields';
+import { ItemsService } from './items';
+
+const systemFields = ['id', 'status', 'sort', 'user_created', 'date_created', 'user_updated', 'date_updated'];
```
We shouldn't assume these system fields exist. People can name them whatever they want

yarick2000 replied:
Yeah, but it will be an alternate way to detect non-translatable fields in translations collection? I've seen in multiple places in the code that system fields were identified just through hardcoded constants...

rijkvanzanten replied:
> I've seen in multiple places in the code that system fields were identified just through hardcoded constants...

That doesn't sound right to me. It's always using the information from the schema object, which is dynamically fetched at the beginning of the request.

yarick2000 replied:
Ok, I will try then to figure out how to do it properly and will fix it

yarick2000 replied:
Actually, here is what I was talking about - this is a piece of code in new-collection.vue that specifying a list of predefined system columns:
![image](https://user-images.githubusercontent.com/13006578/115940619-3e183b80-a470-11eb-8cf7-7df525601346.png)

yarick2000 replied:
So I understand that in my list I should, for example, instead of 'id' use the primary key name. But regarding to other fields - I don't know how to bring them from schema... On client side, as you see, the hardcoded list used.

yarick2000 replied:
Actually, I see that status and sort field probably also can be calculated... But still, what about others?

yarick2000 replied:
Ah, I got it now. For Translations collection the wizard for optional system fields is not used at all.. So ok, I can assume that everything, except of key fields, is translatable, right?

rijkvanzanten commented on api/src/types/query.ts:
```diff
@@ -10,8 +10,9 @@ export type Query = {
-	export?: 'json' | 'csv';
+	export?: 'json' | 'csv' | 'xliff' | 'xliff2';
 	deep?: Record<string, Query>;
+	optional?: Record<string, any>;
```
What does this do?

yarick2000 replied:
The 'optional' is an additional set of parameters that can be used by specific endpoints (and may be just in specific cases). So in order to not pollute type definition for Query with all this stuff I've added just one called 'optional' that may contain different props, depending on current endpoint

rijkvanzanten reviewed (dismissed):
This is all coming together really nicely 😄 

I think I'm going to implement https://github.com/directus/directus/issues/2927 first, and then apply these changes on top of that to officially make XLIFF one of the support formats we can export/import from/to. That'll ensure that we don't end up with split functionality across the endpoints.

@yarick2000 I had a question: Right now, are you exporting the translations based on the _related_ collection? Eg if I'm on `articles` I can export from `articles_translations`? That makes the most sense from an app perspective, but we should make sure we do the export/importing directly on the actual collection instead of through the relationship (not sure if that's the case rn)

rijkvanzanten commented:
Semi-related question. Is XLIFF 1 still used widely? XLIFF 2 has been around since 2014, and it would save a bunch of overhead if we'd just focus on XLIFF 2 exclusively

yarick2000 commented:
> Semi-related question. Is XLIFF 1 still used widely? XLIFF 2 has been around since 2014, and it would save a bunch of overhead if we'd just focus on XLIFF 2 exclusively

Yes, I know, but I need XLIFF 1 for my projects :)

rijkvanzanten commented:
> Yes, I know, but I need XLIFF 1 for my projects :) I guess we can remove it, but wondering then if there is a way to implement it as an extension that will look as part of that UI.... Also I think Xliff 1 still can be popular across organizations...

yaroslav-vysotsky committed:
ARC-4567:

* Moved import to Items controller as per PR suggestion
* Removed unnecessary system fields exclusion

yaroslav-vysotsky committed:
ARC-4567:

* Refactored XliffService basing on request from PR - now it works directly with translations collection

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'fix-4658' of https://github.com/yarick2000/directus into fix-4658

yarick2000 commented:
@rijkvanzanten , I've made the changes you was asking to do: please review. 
Thanks.

yaroslav-vysotsky committed:
ARC-4567:

* Replaced usage of FieldsService with Schema
* Replaced Error with InvalidPayloadException translations and xliff services

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	api/src/controllers/items.ts
#	api/src/middleware/respond.ts
#	api/src/services/index.ts
#	api/src/utils/sanitize-query.ts
#	docs/guides/accessing-data.md

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	api/src/middleware/respond.ts
#	api/src/types/query.ts
#	api/src/utils/validate-query.ts
#	app/src/lang/translations/bg-BG.yaml
#	app/src/lang/translations/en-US.yaml
#	app/src/lang/translations/fi-FI.yaml
#	app/src/lang/translations/pl-PL.yaml
#	app/src/lang/translations/pt-BR.yaml
#	app/src/modules/collections/routes/collection.vue
#	app/src/utils/get-default-interface-for-type/get-default-interface-for-type.ts
#	app/src/views/private/components/export-sidebar-detail/export-sidebar-detail.vue
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
Moved ImportSidebarDetail injection into Layouts code

Fixed Xliff related components in order to match latest changes in Main branch.

yaroslav-vysotsky committed:
Added collectionInfo reset when user navigates across collections

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yarick2000 commented:
Hi @rijkvanzanten , would it be possible to review this branch in the nearest future? It became harder to support this branch in a sync state with Main since more and more changes coming in there....

yarick2000 commented:
> This is all coming together really nicely 😄
> 
> I think I'm going to implement #2927 first, and then apply these changes on top of that to officially make XLIFF one of the support formats we can export/import from/to. That'll ensure that we don't end up with split functionality across the endpoints.
> 
> @yarick2000 I had a question: Right now, are you exporting the translations based on the _related_ collection? Eg if I'm on `articles` I can export from `articles_translations`? That makes the most sense from an app perspective, but we should make sure we do the export/importing directly on the actual collection instead of through the relationship (not sure if that's the case rn)

Ah, I've just realized that I didn't answer to this question. So yes, I'm exporting translations basing on relationships to translations collections. It is possible to make export/import directly from translations collections, but I think it will be very inconvenient for content editors /translators. First, translations collections usually hidden and so in order to make export/import possible CMS admins or developers will have to make them visible, just for that purpose. Second, translators will have to use filters each time in order to select items for specific language for exporting. That can be annoying too...

rijkvanzanten commented:
Yup! Totally agree. That's why I mentioned the difference between app and API 🙂 I think we should do the actual export on the `articles_translations` collection in the API, but have the export widget on the `articles` collection in the app 👍🏻

yarick2000 commented:
> Yup! Totally agree. That's why I mentioned the difference between app and API 🙂 I think we should do the actual export on the `articles_translations` collection in the API, but have the export widget on the `articles` collection in the app 👍🏻

Ah, got it now. Ok, will fix it

yaroslav-vysotsky committed:
ARC-4567:

* Fixed XLIFF export on API side so instead of passing parent collection name the translations collection is used now
* Modified client side accordingly (export-sidebar-detail component)

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	app/src/lang/translations/de-DE.yaml
#	app/src/lang/translations/fr-FR.yaml
#	app/src/lang/translations/id-ID.yaml
#	app/src/lang/translations/ru-RU.yaml

yaroslav-vysotsky committed:
ARC-4567:

* Updated code after getting latest changes from main branch

yaroslav-vysotsky committed:
ARC-4567:

* Fixed XLIFF import on API side so instead of passing parent collection name the translations collection is used now
* Modified client side accordingly (import-import-sidebar-detail component)

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	api/src/types/query.ts

yaroslav-vysotsky committed:
ARC-4567:

* Removed translations service since it's became obsolete

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yarick2000 commented:
> Yup! Totally agree. That's why I mentioned the difference between app and API 🙂 I think we should do the actual export on the `articles_translations` collection in the API, but have the export widget on the `articles` collection in the app 👍🏻
Hi @rijkvanzanten , so I've made those changes, could you please review? Thanks!

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	app/src/lang/translations/ar-SA.yaml
#	app/src/lang/translations/es-419.yaml
#	app/src/lang/translations/es-ES.yaml
#	app/src/lang/translations/it-IT.yaml
#	app/src/lang/translations/tr-TR.yaml
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	app/src/lang/translations/es-CL.yaml
#	app/src/layouts/calendar/calendar.vue
#	app/src/layouts/cards/cards.vue
#	app/src/layouts/tabular/tabular.vue
#	app/src/views/private/components/drawer-item/drawer-item.vue
#	app/src/views/private/components/export-sidebar-detail/export-sidebar-detail.vue
#	package-lock.json

yaroslav-vysotsky committed:
ARC-4567:

Refactored components after getting latest changes from main branch

yaroslav-vysotsky committed:
ARC-4567:

Fixed issue with incorrect path to compiled files - replaced single \ with \\

yaroslav-vysotsky committed:
ARC-4567:

* Fixed issue with incorrect filtering for import endpoint
* Made more refactoring work for import and export components

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

yaroslav-vysotsky committed:
ARC-4567:

* Simplified fix for replacing \ with \\ in  build.js in docs folder

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

# Conflicts:
#	api/src/middleware/respond.ts
#	app/src/lang/translations/nl-NL.yaml
#	app/src/lang/translations/th-TH.yaml
#	app/src/views/private/components/export-sidebar-detail/export-sidebar-detail.vue
#	package-lock.json
#	package.json

yaroslav-vysotsky committed:
ARC-4567:

* Fixed merge artifacts

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

yaroslav-vysotsky committed:
Merge branch 'development' into task/ARC-4567

yaroslav-vysotsky committed:
ARC-4567:

* Removed temporary fix related to that issue: https://github.com/directus/directus/issues/6550 since it was fixed in the main branch

yaroslav-vysotsky committed:
Merge branch 'task/ARC-4567' into fix-4658

rijkvanzanten commented:
Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. 
We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..
